gap/attributes/attr.gi

```diff
@@ -1045,13 +1045,166 @@ function(S)
+        return NambooripadLeqRegularSemigroup(S);
+    else 
+        return MitschLeqSemigroupNC(S);
+    fi;
+end);
+
+InstallMethod(MitschOrderOfSemigroup,
+"for a finite transformation semigroup",
+[IsTransformationSemigroup],
+function(S)
+    local elts, p, func1, func2, out, i, j, regular, D, a;
+    elts := ShallowCopy(AsListCanonical(S));
+    p    := Sortex(elts, {x, y} -> IsGreensDGreaterThanFunc(S)(y, x)) ^ -1;
+    func1 := RegularLeqTransformationSemigroupNC(S);
+    func2 := MitschLeqSemigroupNC(S);
+    out  := List([1 .. Size(S)], x -> []);
+    regular := BlistList([1 .. Size(S)], []);
+    for D in RegularDClasses(S) do
+        for a in D do
+            i := PositionCanonical(S, a) ^ (p ^ -1);
+            regular[i] := true;
+        od;
+    od;
+    for j in [Size(S), Size(S) - 1 .. 1] do
+        if regular[j] then
+            for i in [j - 1, j - 2 .. 1] do
+                if func1(elts[i], elts[j]) then
+                    AddSet(out[j ^ p], i ^ p);
+                fi;
+            od;
+        else
+            for i in [j - 1, j - 2 .. 1] do
+                if func2(elts[i], elts[j]) then
+                    AddSet(out[j ^ p], i ^ p);
+                fi;
+            od;
+        fi;
+    od;
+    return out;
+end);
+
+InstallMethod(MitschOrderOfSemigroup,
+"for a finite regular transformation semigroup",
+[IsRegularSemigroup and IsTransformationSemigroup],
+function(S)
+    local elts, p, func, out, i, j; 
+    elts := ShallowCopy(Elements(S));
+    p    := Sortex(elts, {x, y} -> IsGreensDGreaterThanFunc(S)(y, x)) ^ -1;
+    func := RegularLeqTransformationSemigroupNC(S);
+    out  := List([1 .. Size(S)], x -> []);
+    for i in [1 .. Size(S)] do
+        for j in [i + 1 .. Size(S)] do
+            if func(elts[i], elts[j]) then
+                AddSet(out[j ^ p], i ^ p);
+            fi;
+        od;
+    od;
+    return out;
+end);
+
+InstallMethod(MitschOrderOfSemigroup,
+"for a finite semigroup",
+[IsFinite and IsSemigroup],
+function(S)
+    local i, iso, T, MT, MS, eltsS, eltsT, idx_map, q;
+    iso := IsomorphismTransformationSemigroup(S);
+    T := Range(iso);
+    eltsS := Elements(S);
+    eltsT := Elements(T);
+    idx_map := List([1 .. Size(S)], i -> Position(eltsT, eltsS[i] ^ iso));
+    q := PermList(idx_map);
+    MT := MitschOrderOfSemigroup(T);
+    MS := [];
+    for i in [1 .. Size(S)] do
+        MS[i] := AsSet(OnTuples(MT[i ^ q], q ^ -1));
+    od;
+    return MS;
+end);
+
+InstallMethod(DumbMitschOrderOfSemigroup,
+"for a finite semigroup",
+[IsSemigroup and IsFinite],
+function(S)
+    local func, out, i, j, elts;
+    func := MitschLeqSemigroup(S);
+    elts := Elements(S);
+    out  := List([1 .. Size(S)], x -> []);
+    for i in [1 .. Size(S)] do
+        for j in [1 .. Size(S)] do
+            if i = j then
+                continue;
+            elif func(elts[i], elts[j]) then
+                AddSet(out[j], i);
+            fi;
+        od;
+    od;
+    return out;
+end);
```

Comment:
This should probably not be a separate method. I would suggest adding an optional flag for `MitschOrderOfSemigroup` which indicates the method should use the "dumb" algorithm. Also, if the dumb algorithm is always slower then this bit of code can be removed. Needs some benchmarks.